run_reg_test.py

```diff
@@ -0,0 +1,351 @@
+#!/usr/bin/env python3
+"""
+    Module for running regression tests
+"""
+from pathlib import Path
+import sys
+import argparse
+import textwrap
+import subprocess
+from collections import OrderedDict
+from prettytable import PrettyTable
+
+
+# pylint: disable=wrong-import-position, import-error
+sys.path.insert(
+    0, str(Path(__file__).resolve().parent / "vtr_flow/scripts/python_libs")
+)
+sys.path.insert(0, str(Path(__file__).resolve().parent / "vtr_flow/scripts"))
+from run_vtr_task import vtr_command_main as run_vtr_task
+from vtr import (
+    find_vtr_file,
+    find_vtr_root,
+    RawDefaultHelpFormatter,
+)
+
+# pylint: enable=wrong-import-position, import-error
+BASIC_VERBOSITY = 1
+
+
+def vtr_command_argparser(prog=None):
+    """ Parses the arguments of run_reg_test """
+
+    description = textwrap.dedent(
+        """
+                    Runs one or more VTR regression tests.
+                    """
+    )
+    epilog = textwrap.dedent(
+        """
+                Examples
+                --------
+
+                    Run the regression test 'vtr_reg_strong':
+
+                        %(prog)s vtr_reg_strong
+
+                    Run the regression tests 'vtr_reg_basic' and 'vtr_reg_strong':
+
+                        %(prog)s vtr_reg_basic vtr_reg_strong
+
+                    Run regression tests 'vtr_reg_basic' and 'vtr_reg_strong'
+                    with 8 parallel workers:
+
+                        %(prog)s vtr_reg_basic vtr_reg_strong -j8
+                """
+    )
+
+    parser = argparse.ArgumentParser(
+        prog=prog,
+        description=description,
+        epilog=epilog,
+        formatter_class=RawDefaultHelpFormatter,
+    )
+
+    #
+    # Major arguments
+    #
+    parser.add_argument(
+        "reg_test",
+        nargs="+",
+        choices=[
+            "vtr_reg_basic",
+            "vtr_reg_strong",
+            "vtr_reg_nightly",
+            "vtr_reg_weekly",
+            "odin_reg_full",
+            "odin_reg_syntax",
+            "odin_reg_arch",
+            "odin_reg_operators",
+            "odin_reg_large",
+            "odin_reg",
+            "odin_reg_micro",
+            "vtr_reg_valgrind_small",
+        ],
+        help="Regression tests to be run",
```

Comment:
Should these tests be in a config file, or inferred from the directory structure?

Reply:
This has been fixed